overlord/snapstate/snapmgr.go

```diff
@@ -1182,6 +1183,30 @@ func (m *SnapManager) undoLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func (m *SnapManager) doSwitchSnapChannel(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	snapsup, snapst, err := snapSetupAndState(t)
+	if err != nil {
+		return err
+	}
+
+	// switched the tracked channel
+	snapst.Channel = snapsup.Channel
+	// optionally support switching the current snap channel too, e.g.
+	// if a snap is in both stable and candidate with the same revision
+	// we can update it here and it will be displayed correctly in the UI
+	if snapsup.SideInfo.Channel != "" {
+		snapst.CurrentSideInfo().Channel = snapsup.Channel
+	}
+	m.backend.SwitchSnapChannel(snapsup.Name(), snapsup.Channel)
```

Comment:
we do this sometimes, wondering if it's a bit overkill here though, we check the state at the end anyway (and check the task is there in the other test). if we keep it probably should have a comment that this just for tests